File: app/src/main/java/com/klim/us_stock/di/info/InfoComponent.java
package com.klim.us_stock.di.info;

import com.klim.us_stock.ui.windows.info.InfoFragment;

import dagger.Subcomponent;

@InfoScope
@Subcomponent(modules = {InfoModule.class})
public interface InfoComponent {

    public void inject(InfoFragment fragment);

}

File: src/app/src/main/java/com/klim/us_stock/di/info/InfoComponent.kt
package com.klim.us_stock.di.info

import com.klim.di.ApplicationContextProvider
import com.klim.us_stock.ui.windows.info.InfoFragment
import dagger.Component

@InfoScope
@Component(
    dependencies = [ApplicationContextProvider::class],
    modules = [InfoModule::class]
)
interface InfoComponent {
    fun inject(fragment: InfoFragment?)

    class Initializer private constructor() {
        companion object {
            fun init(appComponent: ApplicationContextProvider): InfoComponent =
                DaggerInfoComponent.builder()
                    .applicationContextProvider(appComponent)
                    .build()
        }
    }
}